game-core/src/main/java/games/strategy/triplea/UnitUtils.java

```diff
@@ -159,15 +159,48 @@ public static int getHowMuchCanUnitProduce(
   }
 
   /**
-   * Currently made for translating unit damage from one unit to another unit. Will adjust damage to
-   * be within max damage for the new units.
+   * Translates attributes and properties from one unit to a collection of units.
+   *
+   * <p>Used when a unit is being transformed, so the old unit is going away and the new units are
+   * taking its place
+   *
+   * <p>Currently, it translates: Hits, Damage, Unloaded units, and Transported units
+   *
+   * <p>Hits and Damage are modified as needed to fit within the limits of the new units. Units will
+   * always have at least 1 hp.
+   *
+   * <p>Unloaded units and transported units are given to the unit that matches stream().findFirst()
    *
    * @return change for unit's properties
    */
   public static Change translateAttributesToOtherUnits(
       final Unit unitGivingAttributes,
       final Collection<Unit> unitsThatWillGetAttributes,
       final Territory territory) {
+
+    final CompositeChange changes =
+        unitsThatWillGetAttributes.stream()
+            .findFirst()
+            .map(
+                receivingUnit -> {
```

Comment:
What would you think of extracting this map to a helper function? AFAIK best practice for lambda's is to keep them one line, and even better to try and make them be method references. That could probably be done here by making a method that accepts the 'receivingUnit' as  an input parameter.